e2e/nodebootstrapping.go

```diff
@@ -0,0 +1,51 @@
+package e2e_test
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/Azure/agentbaker/pkg/agent"
+	"github.com/Azure/agentbaker/pkg/agent/datamodel"
+	"log"
+	"os/exec"
+)
+
+type getNodeBootstrappingFn func(ctx context.Context, nbc *datamodel.NodeBootstrappingConfiguration) (*datamodel.NodeBootstrapping, error)
```

Comment:
actually can we rename this type to nodeBootstrappingFn and then rename getNodeBootstrapping to getNodeBootstrappingFn since its returning a function now rather than node bootstrapping?